feature/permissions/src/main/java/com/google/jetpackcamera/permissions/ui/PermissionsScreenComponents.kt

```diff
@@ -37,41 +37,67 @@ import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.graphics.painter.Painter
-import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.graphics.vector.ImageVector
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import com.google.accompanist.permissions.ExperimentalPermissionsApi
 import com.google.accompanist.permissions.PermissionState
-import com.google.jetpackcamera.R
+import com.google.accompanist.permissions.shouldShowRationale
+import com.google.jetpackcamera.permissions.PermissionEnum
+import com.google.jetpackcamera.permissions.R
 
+/**
+ * Template for a single page for the permissions Screen
+ *
+ * @param permissionEnum a [PermissionEnum] representing the target permission
+ * @param permissionState a [PermissionState] of the target permission
+ */
 @OptIn(ExperimentalPermissionsApi::class)
 @Composable
-fun CameraPermission(cameraPermissionState: PermissionState, modifier: Modifier = Modifier) {
+fun PermissionTemplate(
+    modifier: Modifier = Modifier,
+    permissionEnum: PermissionEnum,
+    permissionState: PermissionState,
+    onRequestPermission: () -> Unit,
+    onSkipPermission: (() -> Unit)? = null,
+    onOpenAppSettings: () -> Unit
+) {
     PermissionTemplate(
         modifier = modifier,
-        permissionState = cameraPermissionState,
-        painter = painterResource(id = R.drawable.photo_camera),
-        iconAccessibilityText = stringResource(id = R.string.camera_permission_accessibility_text),
-        title = stringResource(id = R.string.camera_permission_screen_title),
-        bodyText = stringResource(id = R.string.camera_permission_required_rationale),
-        requestButtonText = stringResource(R.string.request_permission)
+        onRequestPermission = {
+            // if declined by user, must navigate to system app settings to enable permission
+            // todo(b/337338326): open a dialog that tells user they must go to device's app settings to enable permissions
```

Comment:
Can we just change the message in the `PermissionTemplate`'s rationale text to re-iterate that the camera permission is required and change the button text to say something like "Go to app settings"?

Reply:
<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbG12YWdwemZ2dzd2ZG9kMnV0ZDdhcHN6dXAwZHZleGtvbXc3dW84OCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/GSu3kEoodHz8BmdKfD/giphy.gif" width="250" height="250"/>